### DISCUSSION TITLE
e0 calculation

### DISCUSSION THREAD

**Discussion by shenoynikhil** [ACEsuit/mace#126] [Q&A] [answered]
I noticed that there is a [dataset utils function](https://github.com/ACEsuit/mace/blob/44b6da4a5edaa4c3ef867a11728555403d1d475d/mace/data/utils.py#L240) that calculates E0 values. 

I was curious about why this is right as I tried doing this on GEOMDrugs dataset which does not provide e0 values. Post calculating using this method, if I calculate formation energies (`formation energy = total energy - sum of atomization energy`), it is a positive value. Isn't this incorrect as it should be negative?

I'm curious if this is the right way to compute e0 values.

**Comment by davkovacs** [comment, marked as answer]
Hi! 
The utils function calculates an E0 which corresonds to the average atomic energy in the dataset. This means that ca half the structures should have positive and half negative energy compared to the some of the E0-s determiend this way. 
On the other hand if you use the isolated atom energies coming from DFT calculations as the E0-s, than the interaction energy will typically be negative. Though, note that for structures with close approaches, or high pressure phases the interaction energy can be positive even in that case. 

Let me know if this answers your question :)

**Reply by shenoynikhil** [reply]
That helps!
